nixos/modules/virtualisation/virtualbox-guest.nix

```
@@ -111,7 +117,7 @@ in
 
         environment.systemPackages = [ kernel.virtualboxGuestAdditions ];
 
-        boot.extraModulePackages = [ kernel.virtualboxGuestAdditions ];
+        boot.extraModulePackages = lib.optional cfg.use3rdPartyModules kernel.virtualboxGuestAdditions;
```

```suggestion
        boot.extraModulePackages = lib.mkIf cfg.use3rdPartyModules [ kernel.virtualboxGuestAdditions ];
```
otherwise we pollute declerations.